misk/web/@misk/components/src/components/TextHTMLOrElementComponent.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from "react"
+
+/**
+ * <TextHTMLOrElementComponent content={<div>item</div>} length={35} />
+ * <TextHTMLOrElementComponent content={"Test"} length={35} />
+ * <TextHTMLOrElementComponent content={'<a href="http://squareup.com/">Test</a>'} length={35} />
+ *
+ * Renders passed in text, HTML-parseable string, or React element with optional length constraints for text.
+ */
+
+export interface ITextHTMLOrElementProps {
+  content: string | Element | JSX.Element
+}
+
+export const TextHTMLOrElementComponent = (
+  props: ITextHTMLOrElementProps
+): JSX.Element => {
+  const { content }: any = props
+  let FormattedContent: any = content
+
+  if (typeof content === "string" && !content.startsWith("<")) {
+    FormattedContent = <span>{content}</span>
+  } else if (typeof content === "string" && content.startsWith("<")) {
+    FormattedContent = <span dangerouslySetInnerHTML={{ __html: content }} />
+  }
```

Comment:
If we receive a `JSX.Element` here, do we need to wrap it in a `span` too?

Reply:
If it's already a `JSX.Element` then it's ready to be rendered. Wrapping it in an additional `span` can end up messing with `FlexBox` CSS arrangement unfortunately.